site/en/docs/web-platform/view-transitions/index.md

```diff
@@ -649,100 +653,250 @@ addEventListener('click', event => (lastClick = event));
+## Transitions as an enhancement
 
-It's also a bit weird that the code appears to animate `document.documentElement`, only for an option (`pseudoElement`) to change the target of the animation. Fixing this is a larger piece of work, and involves [exposing pseudo-elements as DOM objects](https://drafts.csswg.org/css-pseudo-4/#cssom), so things like `animate` can be called on them directly.
-{% endAside %}
+The View Transition API is designed to 'wrap' a DOM change and create a transition for it. However, the transition should be treated as an enhancement, as in, your app shouldn't enter an 'error' state if the DOM change succeeds, but the transition fails. Ideally the transition shouldn't fail, but if it does, it shouldn't break the rest of the user experience.
 
-## API reference
+In order to treat transitions as an enhancement, take care not to use transition promises in a way that would cause your app to throw if the transition fails.
 
-`document.createDocumentTransition()`
-: Create a new `DocumentTransition`.
+{% Compare 'worse' %}
 
-Instance members of `DocumentTransition`:
+```js
+async function switchView(data) {
+  // Fallback for browsers that don't support this API:
+  if (!document.startViewTransition) {
+    await updateTheDOM(data);
+    return;
+  }
+
+  const transition = document.startViewTransition(async () => {
+    await updateTheDOM(data);
+  });
 
-`const finishedPromise = documentTransition.start(callback)`
-: `callback` is called once the current state of the document is captured.
+  await transition.ready;
+
+  document.documentElement.animate(
+    {
+      clipPath: [`inset(50%)`, `inset(0)`],
+    },
+    {
+      duration: 500,
+      easing: 'ease-in',
+      pseudoElement: '::view-transition-new(root)',
+    }
+  );
+}
+```
 
-    Then, when the promise returned by `callback` fulfills, the transition begins in the next frame. If the promise returned by `callback` rejects, the transition is abandoned.
+{% endCompare %}
 
-    `finishedPromise` fulfills once the transition is complete, or rejects if the transition is abandoned.
+The problem with this example is that `switchView()` will reject if the _transition_ cannot reach a `ready` state, but that doesn't mean that the view failed to switch. The DOM may have successfully updated, but there were duplicate `view-transition-name`s, so the transition was skipped.
 
-`documentTransition.abandon()`
-: Abandon the transition, skipping to the final state.
+Instead:
 
-{% Aside %}
-This API will likely change after the origin trial, to improve usability.
-{% endAside %}
+{% Compare 'better' %}
+
+```js
+async function switchView(data) {
+  // Fallback for browsers that don't support this API:
+  if (!document.startViewTransition) {
+    await updateTheDOM(data);
+    return;
+  }
+
+  const transition = document.startViewTransition(async () => {
+    await updateTheDOM(data);
+  });
+
+  animateFromMiddle(transition);
+
+  await transition.domUpdated;
```

Comment:
```suggestion
  await transition.finished;
```

Is finished a better fit if you want this function to return once the DOM has flipped to the new state, either with the animations finishing or directly?

Reply:
I think there are cases for one or the other. I'll add a note/comment.